.github/workflows/check.yml

```
@@ -1,57 +1,39 @@
 name: Check Set-Up, Build, Test
 
-# Controls when the action will run. 
+# Controls when the action will run.
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
-    branches: [ master ]
+    branches:
+    - master
   pull_request:
-    branches: [ master ]
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  check-build:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-20.04
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
+  linting:
+    name: "Check rustfmt"
+    runs-on: ubuntu-latest
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-
-      - name: Set-Up
-        run: sudo apt install -y cmake pkg-config libssl-dev git build-essential clang libclang-dev curl
-
-      - name: Install Rustup
-        run: |
-          curl https://sh.rustup.rs -sSf | sh -s -- -y
-          source ~/.cargo/env
-
-      - name: Check Build
-        run: |
-          SKIP_WASM_BUILD=1 cargo check --release --features all-nodes
-
-  test-coverage:
-    runs-on: ubuntu-20.04
-
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Rust
+      run: rustup show
+    - name: rustfmt
+      run: cargo fmt -- --check
+  build:
+    name: "Build release"
+    runs-on: self-hosted
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Set-Up
-        run: sudo apt install -y cmake pkg-config libssl-dev git build-essential clang libclang-dev curl
-
-      - name: Install Rustup
-        run: |
-          curl https://sh.rustup.rs -sSf | sh -s -- -y
-          source ~/.cargo/env
-          rustup update stable
-          rustup component add llvm-tools-preview
-          cargo install grcov
-          cargo install rust-covfix
-
-      - name: Test coverage
-        run: |
-          sh ./.maintain/test-coverage/test-coverage.sh
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Rust
+      run: rustup show
+    - name: Build
+      run: |
+        cargo build --release --all
```

```suggestion
        cargo build --release --locked --all-features
```
We may want to consider further limiting this to exclude the runtime-benchmarks features and anything else that might be low value but time consuming. But this is a good start regardless.